### ISSUE TITLE
IT_NO_SUCH_ELEMENT false-positive

### ISSUE THREAD

**Issue by boris-petrov** [spotbugs/spotbugs#1549]
```java
public static <T> Stream<List<T>> zip(List<Stream<T>> list) {
	List<Iterator<T>> iterators = list.stream().map(Stream::iterator).collect(Collectors.toList());
	Iterator<List<T>> iteratorR = new Iterator<>() {
		@Override
		public boolean hasNext() {
			return iterators.stream().allMatch(Iterator::hasNext);
		}

		@Override
		public List<T> next() {
			return iterators.stream().map(Iterator::next).collect(Collectors.toList());
		}
	};
	boolean parallel = list.stream().anyMatch(Stream::isParallel);
	Iterable<T> iterable = () -> iteratorR;
	return StreamSupport.stream(iterable.spliterator(), parallel);
}
```

This code gives a `IT_NO_SUCH_ELEMENT` error. This is not correct because in my `next` call I do call `Iterator::next`. It's probably a bit difficult for SpotBugs to figure that out though...
